compute_tools/src/extension_server.rs

```diff
@@ -174,6 +174,20 @@ pub async fn get_available_libraries(
         );
     }
 
+    // TODO: shouldn't we filter this by .so?
+    /*
+    right now we are getting weird stuff like
+    "pg_isolation_regress": RemotePath("v14/lib/pgxs/src/test/isolation/pg_isolation_regress"),
+    Makefile.shlib": RemotePath("v14/lib/pgxs/src/Makefile.shlib")
+    which we should not try to download...
+
+    the expected format is
+    "timescaledb-2.10.1.so": RemotePath("v14/lib/timescaledb-2.10.1.so")
+
+    Q: do we care about files like this?
+    libecpg.so.6": RemotePath("v14/lib/libecpg.so.6")
+    libpgtypes.so.3": RemotePath("v14/lib/libpgtypes.so.3")
+     */
```

Comment:
yes we do

we need to change "enforce_so_end" function to handle this case -
don't add .so suffix if it is already present somewhere in the filename

Reply:
```rs
// appends an .so suffix to libname if it does not already have one
fn enforce_so_end(libname: &str) -> String {
    if !libname.contains(".so") {
        format!("{}.so", libname)
    } else {
        libname.to_string()
    }
}```
Does this  alternate logic make sense?

Reply:
Also, if the user types `LOAD libpgtypes`
Do we need to find `libpgtypes.so.3` and download that?